### ISSUE TITLE
Nullable warning with implicit operator and AllowNull

### ISSUE THREAD

**Issue by grokys** [dotnet/roslyn#43225]
**Version Used**: 

VS 16.5.2

**Steps to Reproduce**:

```
public class Foo<T>
{
    [AllowNull] private T _value;

    public Foo([AllowNull] T value)
    {
        _value = value;
    }

    public static implicit operator Foo<T>([AllowNull] T value) => new Foo<T>(value);
}

object? o = null;
Foo<object> f = o; // CS8604
Foo<object> g = new Foo<object>(o); // This works

```

**Expected Behavior**:

```
Foo<object> f = o; // CS8604
Foo<object> g = new Foo<object>(o); // This works
```

The two statements should be equivalent and not warn

**Actual Behavior**:

```
Warning	CS8604	Possible null reference argument for parameter 'value' in 'Foo<object>.implicit operator Foo<object>(object value)'.
```

**Comment by mikernet** [comment]
Also related:
```c#
var h = (Foo<object>)o; // CS8600 
```

**Comment by RikkiGibson** [comment]
I believe this was fixed by #47555. [SharpLab](https://sharplab.io/#v2:EYLgtghglgdgNAExAagD4GIYFcA2OLA4CmABETAcQLABQAAgAwl0CMAdACJQQDmMA9gGcALlADGgtgGF+CIgEEKOAJ6CoggNy1tNfsABWRMcID8JfiQC8JbHi00AYv34AePYeMA+EgDMr5jRIAeiCSKQBlAA4ANgYAFlonV3cjYW8efxgiAHcSJLcDVM8ACn4ASkCQkgAVAAt1Emz+ACcAa0EdWjoAZmYAJjznF2rPWgBvWhIpkgBteTx+bIA5XBwAXRIAB2aoADcIYVJqkgB9fZwsIntJ6Z7B/mK5heXVjePzy7KbqYmaaf/Th9SNYgfZ/gBfHT/O6saIkKBgTY4cRQYTmTZEZoHFr3YYlJ44RYrPBvEhAspWbxZXL5EbFcn2SE0IA=)
